### ISSUE TITLE
Issues with New "firefly" G6 transmitter (serial# 8G... or 8H...)

### ISSUE THREAD

**Issue by cbadams7** [NightscoutFoundation/xDrip#1084]
Recieved two new transmitters for my dexcom G6 with serial numbers beginning in 8G. Attempted multiple times to follow instructions listed here https://androidaps.readthedocs.io/en/latest/EN/Configuration/xdrip.html but with no success. I will sometimes recieved a few minutes of reading but eventually xdrips+ will no longer report current values. Please help! Really need this to work for an upcoming triathlon!

**Comment by zothen** [comment]
Same for me.

**Comment by jamorham** [comment]
Which xDrip+ version are you using? If not very recent please switch to Alpha update channel and try that.

**Comment by xannasavin** [comment]
So I have just started today my 8G Transmitter with the latest Alpha version of xDrip+ and in the beginning, for about 3hrs it collected the date and then suddenly stopped. 
The Dexcom Receiver still works, but xDrip does not connect since hours. restarted phone, bluetooth, collector and everything but nothing works.
With the 80 Transmitter used till today it worked quite well, so the problem has to be with the 8G one.

**Comment by xannasavin** [comment]
ok, so after 3hrs - just after this comment - I tried a step I read somewhere else and now I have at least 1 new data. 
On the system status screen I clicked on "forget device" and restarted the data collector. 
I also got some backfill of data, which is nice, though it was pretty annyoing to not have any update for 3hrs.
Not sure what more information you need for bug tracing?

**Comment by jamorham** [comment]
@xannasavin which version of android are you using?

**Comment by zothen** [comment]
I'm a new xDrip user also with a brand new 8G G6 transmitter, and I can't
keep a reliable connection.

My phone is Pixel 4 XL with Android 10.

**Comment by zothen** [comment]
To elaborate, the main problem for me is if I have one connection problem
for any reason, it won't automatically reconnect.  Restarting the collector
never works.  I have to forget the device and restart the collector and
only after many tries will it reconnect.

I've tried all of the G5/G6 debug options with no luck.  Have also tried
some of the Bluetooth debug options as well.

**Comment by zothen** [comment]
"only after many tries"

I meant only after I repeat my actions of forgetting the device and
restarting the collector will it final reconnect and work for some limited
time.

**Comment by jamorham** [comment]
@zothen I think this is an android 10 issue unrelated to the type of transmitter

**Comment by cbadams7** [comment]
@jamorham I'm running 9, updating to 10 right now

**Comment by xannasavin** [comment]
![Screenshot_20191113_003823_com android settings](https://user-images.githubusercontent.com/48352696/68695588-ee142480-05ad-11ea-9957-f139690b2a2f.jpg)
Thats the complete information about the phone

**Comment by xannasavin** [comment]
![Screenshot_20191113_004046](https://user-images.githubusercontent.com/48352696/68695775-43e8cc80-05ae-11ea-8596-3736d7f992d5.jpg)
Here is the information

**Comment by Dainem21** [comment]
Hello,

if I see it correctly I already got the new transmitter. But I have no issues using a Sony Xperia XZ1 Android 9. Only thing special. It has no service. I use it offline without a SIM Card.

My procedure to start the sensor:
1) Stop the old one and wait until it is stopped. 
               Queue must be empty under status.

<This is special, no need if only xdrip is used>
_I have an iPhone 8 with the Dexcom app, which I use as backup. Bluetooth is forbidden for Dexcom. Therefor I stop the sensor in the Dexcom app, too. I switch on Bluetooth for the Dexcom App and start the sensor.  Without waiting for the App to connect I switch of Bluetooth and block it again for the Dexcom app._ 
<xDrip+>
2) I start a new sensor in xDrip and enter Code. 
3) Wait for the warmup to finish. During this yDrip will connect to the sensor and you see the countdown.
4)Working
<This is special, no need if only xdrip is used>
  _The iPhone must stay disconnected to the sensor during its warm up. If the warm up is done, I can switch between xDrip on Android and Dexcom on the iPhone. Of course only one at the same time_

Please find my xDrip settings here:
![Screenshot_20191112-191821](https://user-images.githubusercontent.com/41136910/68700028-85648200-0584-11ea-8260-58cb61cc8e7e.png)
![Screenshot_20191112-192037](https://user-images.githubusercontent.com/41136910/68700037-8ac1cc80-0584-11ea-937f-58075d968cf8.png)
![Screenshot_20191112-192046](https://user-images.githubusercontent.com/41136910/68700050-8f868080-0584-11ea-870a-4a7a8add88af.png)
![Screenshot_20191112-192103](https://user-images.githubusercontent.com/41136910/68700077-9c0ad900-0584-11ea-9662-5661e8b8198b.png)
![Screenshot_20191112-192111](https://user-images.githubusercontent.com/41136910/68700082-9e6d3300-0584-11ea-8fa8-33285a633778.png)
![Screenshot_20191112-192119](https://user-images.githubusercontent.com/41136910/68700091-a200ba00-0584-11ea-9c9f-6edb2c41a07b.png)
![Screenshot_20191112-192132](https://user-images.githubusercontent.com/41136910/68700101-a4fbaa80-0584-11ea-8029-fa019e7bf3fe.png)
![Screenshot_20191112-192152](https://user-images.githubusercontent.com/41136910/68700112-ac22b880-0584-11ea-82b5-c3664acc45b3.png)
![Screenshot_20191112-192203](https://user-images.githubusercontent.com/41136910/68700126-b2189980-0584-11ea-8df4-886ce3515af2.png)
![Screenshot_20191112-192212](https://user-images.githubusercontent.com/41136910/68700135-b3e25d00-0584-11ea-9d3b-66920ebaf3e1.png)
![Screenshot_20191112-192223](https://user-images.githubusercontent.com/41136910/68700139-b644b700-0584-11ea-893a-e5a1ec07c9ac.png)
![Screenshot_20191112-192234](https://user-images.githubusercontent.com/41136910/68700143-b775e400-0584-11ea-8088-f089b655be10.png)
![Screenshot_20191112-192240](https://user-images.githubusercontent.com/41136910/68700150-b9d83e00-0584-11ea-864a-f75f384b708a.png)


### **Here in English spelling:**

![Screenshot_20191112-194602](https://user-images.githubusercontent.com/41136910/68700697-9e216780-0585-11ea-87a6-5756d4606a49.png)
![Screenshot_20191112-194611](https://user-images.githubusercontent.com/41136910/68700699-9eb9fe00-0585-11ea-9ae5-60be10acf0f3.png)
![Screenshot_20191112-194619](https://user-images.githubusercontent.com/41136910/68700700-9eb9fe00-0585-11ea-9fff-9f4dfaa4f7af.png)
![Screenshot_20191112-194640](https://user-images.githubusercontent.com/41136910/68700702-9eb9fe00-0585-11ea-80d8-90a4e632e680.png)
![Screenshot_20191112-194651](https://user-images.githubusercontent.com/41136910/68700704-9eb9fe00-0585-11ea-8e6a-f7cc7f367a96.png)
![Screenshot_20191112-194700](https://user-images.githubusercontent.com/41136910/68700705-9eb9fe00-0585-11ea-9294-9ea9242e99d0.png)
![Screenshot_20191112-194706](https://user-images.githubusercontent.com/41136910/68700706-9f529480-0585-11ea-9f24-e582201c5532.png)
![Screenshot_20191112-194420](https://user-images.githubusercontent.com/41136910/68700707-9f529480-0585-11ea-9ee3-0d6c9086d4ae.png)
![Screenshot_20191112-194429](https://user-images.githubusercontent.com/41136910/68700708-9f529480-0585-11ea-9970-6f501f1c55bd.png)
![Screenshot_20191112-194532](https://user-images.githubusercontent.com/41136910/68700709-9f529480-0585-11ea-8c8e-58124a06e3ba.png)

**Comment by cbadams7** [comment]
Each time I restart collector and forget device within System Status during the next scheduled bluetooth connection I'll receive a reading with 2 hours of back fill. However no connection made and no readings are received until I restart collector and forget device again
![Screenshot_20191112-134250_xDrip+](https://user-images.githubusercontent.com/50087070/68713663-3ce0a080-0553-11ea-869d-1c60517e9276.jpg)
![Screenshot_20191112-134734_xDrip+](https://user-images.githubusercontent.com/50087070/68713667-3eaa6400-0553-11ea-9021-ff95ec734bcf.jpg)

**Comment by xannasavin** [comment]
So since the issue yesterday I did not have connection issues yet. maybe it was a strange hickup, lets see

**Comment by xannasavin** [comment]
So right now Problem again: I guess I went out of reach for short time and now getting since 2hrs no data

**Comment by zothen** [comment]
FYI, I've been using a nightly build from a couple of days ago:

version d7689af-2019.11.02

**Comment by xannasavin** [comment]
> Each time I restart collector and forget device within System Status during the next scheduled bluetooth connection I'll receive a reading with 2 hours of back fill. However no connection made and no readings are received until I restart collector and forget device again
Exaxtly the same for me. 
I installed the dexcom app and it does not seem to have any Problems.

**Comment by xannasavin** [comment]
So for testing I installed the latest dexcom app and it runs smooth without Problems. So they seem to have changed something with the 8g that causes connection issues with xdrip

**Comment by xannasavin** [comment]
@jamorham anything new? :)

**Comment by zothen** [comment]
I found what might be a fix for me.  I've been up and running for 24 hours
straight including being away from my phone for 20+ minutes and it still
reconnects just fine.

Under settings->g5/G6 debug settings, I've disable "Allow OB1 unbonding"
and "Allow OB1 initiate bonding".  I'm not sure which actually leads to the
positive change, but I suspect the latter because I had already once tried
the former.

I previously always got bonding errors when connecting.  I'll have to
re-check that now with these new settings.

**Comment by xannasavin** [comment]
Thank you @zothen - I gave that setting a try, I even made a clean re-install of xDrip, taking those settings into account after setup. Unfortunately it did not help. I get the first reading and that's it. 

Batterie settings are completely off, so no optimization, also after the re-install. But that's was also the case before.

This is super anoying, the original Dexcom app (of course de-installed for the xDrip test) is a hell, giving every 5 minutes alarm on high or low sugar. Can't tell it to shut the f*** up (I know it's high and it takes some time till the insulin kicks in, so leave me alone). Didn't even know how convenient xDrip's alarm snoozing is till now...

So long story short: xDrip doesn't work at all with the new 8g for me, the Dexcom app does not have any problems (besides the App itself is shitty af...). So I guess there must be something between the transmitter and xDrip...

It would be really awesome if someone could take a look into this. Last Dev comment was 5 days ago and the last update beginning of the month - I can't believe I am the only one facing this issues?
Of course I know the devs are not doing this fulltime and I really appreciate their work!

**Comment by zothen** [comment]
@xannasavin, I need to correct something I previously said.

Go back to the default settings (may need to clear app storage data to as
the app doesn't provide a method to so this unfortunately) and then make a
connection with your G8 (this takes me a few tries sometimes).  After it
makes a connection and downloads data, then uncheck the "Allow OB1
unbonding".   If you lose connection again sometime later, you'll want to
re-check this box before manually trying to re-check.

The method above seems to allow me to connect for the longest time (maybe
24 hours).

Ignore my comment about OB1 initiate bonding.

**Comment by cbadams7** [comment]
@xannasavin I'm having the same issue except I cant run the dexcom app on my phone, so I'm stuck with POS receiver. Would love some help!

**Comment by nshtg** [comment]
@cbadams7 
Why can't you run the Dexcom app on your phone? You can build yourself a custom version that ignores the silly device check and should work on most Android phones just fine.

https://www.reddit.com/r/diabetes/comments/c00wgk/oc_build_your_own_dexcom_app_update_bugfixes_and/

**Comment by cbadams7** [comment]
@nshtg 
much appreciated! I'll give it a try. A functional xdrip would be preferable but this will work for now!
Thanks again!

**Comment by xannasavin** [comment]
So I even tried with an galaxy s7 edge phone with no Apps installed at all, but I cant get a connection with xdrip. Dexcom app is running fine, so I doubt the issue is hardware/phone related...

**Comment by xannasavin** [comment]
Btw @cbadams7 Huawei is supported by dexcom since the last Update if this was your issue - maybe other phones too

**Comment by zothen** [comment]
I uninstalled and reinstall the app after inserting a new G6 sensor and
then could not get xDrip to work properly not matter what settings I
tried.  It would only ever grab data on the initial connection and never
again.  I realized the trick was to use the Dexcom receiver first to
connect to the sensor.  I could then power it off and connect fine with the
xDrip app.  I still have the "Allow OB1 bonding" unchecked, but I'm not
sure if that is needed/helpful at this point.

**Comment by nshtg** [comment]
I have the mentioned errors in log for every reading too. Getting values works just fine though.

![Screenshot_20191122-233514](https://user-images.githubusercontent.com/5219415/69477017-6f0ac180-0de1-11ea-9b96-f6218001b65f.jpg)

**Comment by xannasavin** [comment]
so, don't want to party to early, but it looks like with the nightly version from 22.11. the issue is fixed. 
will give tomorrow final feedback.

**Comment by xannasavin** [comment]
ok, after 8hrs the connection was lost and didnt connect the past 1,5hrs...

**Comment by xannasavin** [comment]
uploaded the log btw... let's see

**Comment by Cagier** [comment]
I'm getting very similar errors in the logs and long dropouts between 30 minutes and 2 hours.  Often a phone restart will fix, maybe followed by a collector restart but last time I had to unpair and reboot many times before it finally connected.  Very frustrating!  I've been using xDrip for many years but sometimes I wish we were back on the G4!

**Comment by xannasavin** [comment]
So for me the current sensor works since a few days with unoticable Breaks. I have the ob1 unbound settings mentioned above unchecked.
The thing I did different then the last times: I started the sensor with the dexcom receiver only. Xdrip hooked up after a while itself without entering a Code. Not sure if that did the Trick, I will check with the next sensor.

**Comment by xannasavin** [comment]
Ah I use the latest nightly from 5th december

**Comment by wardc99** [comment]
Are there any updates? We just started our first 8H sensor and we got readings for about 24 hours and now nothing. I've updated settings, I'm using the latest nightly build. Nothing is working. Is it even possible to use xDrip with these transmitters? What a huge bummer!!!!!!!!! I've never had a problem with xDrip before. What other options exist for uploading to a heroku/nightscout site? Thanks.

**Comment by xannasavin** [comment]
I hoped it gets better with the 8H transmitter which is next for me :(
This is really annoying...

**Comment by sweetafro** [comment]
Similar problems here. Tried running an 8H transmitter. Unfortunately I was not in a spot to capture and post logs, etc. I am running Android 9. 

I was running latest nightly build from 29 Jan. Had initial connection to the 8H and then started to be intermittent. Had to give that up due to traveling schedule and need for solid readings.

That 8H Transmitter now only connects to the Dexcom Receiver. It seems that somehow, the transmitter did not release its second bluetooth slot, creating a pairing problem. No luck even after restarting  phone, factory resetting phone, faraday bagging the transmitter and more of the tricks found on various websites. 

Other observation: So far, in my experience, I've found if you stop a G6 sensor from the Dexcom Receiver, there is no way to restart it, so try to avoid that option until the last possible moment if you can. 

If/when I have another go at this with an 8H, I will share logs, etc.

(Thanks so much to the xdrip team for all that you do. Simply amazing.)

**Comment by xannasavin** [comment]
So switched yesterday to the 8H transmitter, got readings through the night, ca. 8hrs, and afterwards no readings anymore. Logs sent per Mail. Makes no sense, since it did read for 8hrs. Dexcom receiver and app without problems (installed the Dexcom app again now, deinstalled before xDrip of course).

**Comment by xannasavin** [comment]
@jamorham this issue is open for really long - could you please have a look? I would really like to get back to using xDrip...

**Comment by nshtg** [comment]
I am on my second 8H transmitter and despite the errors that show up in the log it works flawlessly.

**Comment by xannasavin** [comment]
but where is the culprit then? I also heared from others, that they don't experience problems. but I also got a lot of feedback with issues.
So to summarize once more the setup I'm using:
Huawei P20 Pro
Android 9
EMUI 9.1.0
Build: 9.1.0.372(C782E12R1P11)
xDrip Nightly Build 20th Feb 2020

Sensor: 8Hxxx

What else is needed?

As said I got readings the whole night, just stoping in the morning after about 8hrs and not coming back. Log was sent per Mail...

**Comment by wardc99** [comment]
I'm glad it's working for some people. It's not working for a lot of us. Are there any solutions yet? We have Android 9 on a Google Pixel, most recent nightly build. Ours works for about 12 hours then stops connecting.

**Comment by sweetafro** [comment]
@nshtg Are you running the Dexcom receiver with your 8H transmitter at all? Or only an android device (phone or watch) with xdrip?

While it doesn't help those of us that really, really want to get back on xdrip with our firefly transmitters, this may not be an xdrip-specific issue. The culprit might, in fact, be buggy firefly transmitter bluetooth connectivity issues.

FWIW: a new 8H failed following the Dexcom process, using their phone app and receiver. Perhaps in their valiant attempts to lock down the transmitter and discourage unofficial 3rd party apps, they have created their own problems? A bit like creating home security that works so well, the bad people can't get in but you also can't get out. 

8H failure using Dexcom's receiver and app were all similar to issues described here and on the Dexcom reddit community - transmitter connects to Dexcom receiver fine but loses connection to the phone running the Dexcom phone app and cannot re-connect. It seems like there's something buggy with the firefly transmitter's bluetooth to the phone but not to their receivers. Could their receiver lock on the first connection and the second bluetooth connection to the phone is the flaky one or the receiver connection is somehow interfering? They asked for the transmitter back for troubleshooting. Standard protocol for them. Also, they're now up to 8K transmitter series and will run out of letters before we know it.

Maybe there's a nugget in the above that can help in troubleshooting.

Hopefully this will get sorted sooner rather than later.

**Comment by xannasavin** [comment]
so for me the Dexcom App is working just fine without any interupts, even if connected as third "party" (after xDrip stopped working and got de-installed). Of course the Dexcom app was not on my phone during the usage of xDrip. So I don't suspect any bluetooth issues with the transmitter to be the problem...

**Comment by sweetafro** [comment]
Good to know. I have seen some online speculation that there was a bad lot of sensors (and possibly transmitters) but that is purely speculation.

**Comment by wardc99** [comment]
the 8H that we could not get to work with xDrip is working fine with Share. but Share sucks.

**Comment by sweetafro** [comment]
So true about share.I may have just had a bad transmitter. I guess it's back to xdrip troubleshooting.

**Comment by xannasavin** [comment]
and thread is dead again :(

**Comment by liamawhite** [comment]
FWIW, I changed to an 8H and it did the only connect once before dropping. When I upgraded to the 20th Feb build it fixed the issue and has been working fine for 3+ days now.

**Comment by xannasavin** [comment]
New Sensor, same bad luck. This time only got 5hrs of reading before it stopped.
@jamorham I did just send the logs to you.
I followed the settings and intsructions for the new sensor of this post
https://androidaps.readthedocs.io/en/latest/CROWDIN/de/Configuration/xdrip.html#neuer-sensor
and installed the latest nightly https://github.com/NightscoutFoundation/xDrip/releases/tag/2020.03.11
![Screenshot_20200315_095541_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/48352696/76698542-80cfe780-66a4-11ea-92cf-faeff4b43642.jpg)
![Screenshot_20200315_100126_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/48352696/76698543-84fc0500-66a4-11ea-93fc-fc69f7f3edb9.jpg)
![Screenshot_20200315_095528_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/48352696/76698544-86c5c880-66a4-11ea-874f-50d4ecb299a0.jpg)

**Comment by xannasavin** [comment]
ah, and the Dexcom app that I now installed again this morning got readings within minutes btw

**Comment by adanvdo** [comment]
I've been running an 8G transmitter for a month with no connection issues. The only problem I face is I can't restart any more

**Comment by mefathy** [comment]
I had the same issue and managed to resolve it by disabling battery optimization for xdrip+ and all Bluetooth system apps.

**Comment by sweetafro** [comment]
After using a new 8H transmitter, mine has been working fine. However, I had some intermittent flakiness with my auto bluetooth interfering. Thus far, the 8H and my phone reconnect shortly after that interference. Thanks for the battery optimization tip @mefathy That might just fix that little annoying issue for me. Otherwise, my former problems on this issue are gone.

**Comment by LucaMex** [comment]
I have an expired 8G transmitter I wanted to reset (ran out of transmitters); since I've not used xDrip before I don't know if I'm doing it correctly etc. Is there any guide to follow? The transmitter reset is substantially undocumented.
(and is there any way not to have to use the microphone to try an hard reset? google really likes understanding italian words instead of 'transmitter' making the process even worse)

**Comment by mk886714** [comment]
Sorry 8H can not be "hard resetted"
Software in Transmitter cant be resetted

**Comment by Navid200** [comment]
Since the November 27, 2020 xDrip release was issued, we have had success with all different handsets.  Instructions can be found here:
https://www.facebook.com/groups/xDripG5/permalink/4838757582864365

If you don't like to use facebook, you can ask in the Gitter room.

I like to close this issue.
Please respond if there is a reason not to.

**Comment by adanvdo** [comment]
> Since the November 27, 2020 xDrip release was issued, we have had success with all different handsets. Instructions can be found here:
> https://www.facebook.com/groups/xDripG5/permalink/4838757582864365
> 
> If you don't like to use facebook, you can ask in the Gitter room.
> 
> I like to close this issue.
> Please respond if there is a reason not to.

Can you clarify what you have had success with? I would think that if you are required to remove the transmitter in order to restart the sensor than there is still an issue here. restarting at 10 days has not worked in a long time and the option for auto restarting was disabled due to the firmware not being compatible. I have not seen any commits with changes to address this. 

The issue is with transmitters starting with 8G and after. I have an 8K that also cannot be restarted in the app

**Comment by xannasavin** [comment]
I have since quite some time a stable connection to the Transmitter. For me changing the smartphone did the Job. Neither my old Huawei P20 Pro nor an even older Samsung Galaxy S7 Edge worked. The P30 Pro works. No clue, if it's due to Android Version or any other software/hardware spec.
